ui/src/components/Tree/index.tsx

```diff
@@ -0,0 +1,147 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+import React, { useState, useCallback, useEffect } from "react"
+import Row from "../../scenes/Schema/Row"
+import styled from "styled-components"
+
+export type TreeNodeKind = "column" | "table" | "folder"
+
+export type TreeNodeRenderParams = {
+  toggleOpen: () => void
+  isOpen: boolean
+}
+
+export type TreeNodeRender = ({
+  toggleOpen,
+  isOpen,
+}: TreeNodeRenderParams) => React.ReactElement
+
+export type TreeNode = {
+  name: string
+  kind?: TreeNodeKind
+  render?: TreeNodeRender
+  initiallyOpen?: boolean
+  wrapper?: React.FunctionComponent
+  onOpen?: (leaf: TreeNode) => Promise<TreeNode[]> | undefined
+  children?: TreeNode[]
+}
+
+const Ul = styled.ul`
+  padding: 0;
+  margin: 0;
+`
+
+const Li = styled.li`
+  list-style: none;
+  padding: 0 0 0 1rem;
+`
+
+const WrapWithIf: React.FunctionComponent<{
```

Comment:
I think this could be moved to `components` as it's generic

Reply:
done, moved under `components/WrapWithIf`